core/src/tag_utils.rs

```diff
@@ -52,15 +52,12 @@ impl SwfMovie {
     }
 
     /// Utility method to construct a movie from a file on disk.
+    #[cfg(any(unix, windows, target_os = "redox"))]
     pub fn from_path<P: AsRef<Path>>(path: P, loader_url: Option<String>) -> Result<Self, Error> {
-        let mut url = path.as_ref().to_string_lossy().to_owned().to_string();
-        let cwd = std::env::current_dir()?;
-        if let Ok(abs_url) = url_from_relative_path(cwd, &url) {
-            url = abs_url.into();
-        }
-
+        let path = path.as_ref().canonicalize()?;
+        let url = Url::from_file_path(&path).unwrap();
```

Comment:
Should we use `unwrap_or_default` for the URL here, or at least return an error instead of unwrap? I ask because we have hit panics in this URL<->path conversion in the past (#3095).

Reply:
Ack. An error is returned now.